examples/flutter_firebase_login/packages/authentication_repository/lib/src/authentication_repository.dart

```diff
@@ -74,8 +129,8 @@ class AuthenticationRepository {
         email: email,
         password: password,
       );
-    } on Exception {
-      throw SignUpFailure();
+    } on FirebaseAuthException catch (e) {
```

Comment:
I don't think we should remove `Exception` as it is a possibility that Firebase some time fails with a generic error, should be more like:
```dart
} on FirebaseAuthException catch (e) {
  throw SignUpWithEmailAndPasswordFailure.fromCode(e.code);
} on Exception {
  throw SignUpWithEmailAndPasswordFailure('Unknown');
}
```
What do you think?